RadarSDK/RadarAPIClient.m

```diff
@@ -283,13 +318,22 @@ - (void)trackWithLocation:(CLLocation *_Nonnull)location
 
     BOOL replaying = options.replay == RadarTrackingOptionsReplayAll && replayCount > 0 && !verified;
     if (replaying) {
-        NSMutableArray *replaysArray = [RadarReplay arrayForReplays:replays];
-        [replaysArray addObject:params];
 
-        url = [NSString stringWithFormat:@"%@/v1/track/replay", host];
+        [[RadarReplayBuffer sharedInstance] flushReplaysWithCompletionHandler:params completionHandler:^(RadarStatus status, NSDictionary *_Nullable res) {
+            if (status != RadarStatusSuccess) {
+                [[RadarLogger sharedInstance] logWithLevel:RadarLogLevelDebug message:[NSString stringWithFormat:@"Failed to flush replays"]];
+                [[RadarDelegateHolder sharedInstance] didFailWithStatus:status];
+            } else {
+                [[RadarLogger sharedInstance] logWithLevel:RadarLogLevelDebug message:[NSString stringWithFormat:@"Succeded in flushing replays"]];
+                [RadarState setLastFailedStoppedLocation:nil];
+                [RadarSettings updateLastTrackedTime];
+            }
+
+            [[RadarDelegateHolder sharedInstance] didFailWithStatus:status];
```

Comment:
Should just need the one `didFailWithStatus` from above in `if (status != RadarStatusSuccess)`

Reply:
we need just one, but this is the one we should keep

current behavior is that even if `RadarStatusSuccess`, we fall through to `[[RadarDelegateHolder sharedInstance] didFailWithStatus:status];` when calling `/track/replay`

Reply:
hm is that current behavior intentional though? Seemed like that was the "Radar error received" logs even though no error

Reply:
Ok, I actually figured out how to get toolkit working here while making this behave in a more standard manner. https://github.com/radarlabs/radar-sdk-ios/pull/274/commits/a44b46eb4710b4a67f66f6d40c04b195a290fb90